### PR TITLE
Add prop onInputTextChanged on GiftedChat API and help fix #286

### LINKED ISSUE #286: Strange behavior with onChange in overridden Composer for typing indicator for that user
#### Issue Description

By using the`renderComposer` to access the onChange prop it will disable inherited props like `renderSend` and `this.props.text` and the growing height of the TextInput.

#### Steps to Reproduce / Code Snippets
```javascript
  renderComposer(props) {
    return (
      <Composer {...props} onChange={() => this.handleChangeText()} />
    )
  }
```

#### Expected Results

Composer remains intact and onChange sends back event information to be able to fire off that the user is typing.

#### Additional Information

* React Native version: 0.36.0
* react-native-gifted-chat version: 0.0.6
* Platform(s) (iOS, Android, or both?): iOS

### PR DESCRIPTION
**Pull request by jayshah123**
This introduces a new prop : onInputTextChanged on GiftedChat API,
which is a callback function that will give whatever text is present currently in Composer back to user.
Composer has onChange prop but it is kind of consumed by onType function in GiftedChat.js.
This will also help fix #286, where you see GiftedChat.js personally uses onChange,
and the behaviour such as height calculation would have to be performed by library user (which right now happens in onType in GiftedChat.js) 
This prop gives users a simpler interface to track change in input text, with all other features intact.
Kindly guide with suggestions/comments wherever feels necessary.
Thank you.

### CODE REVIEW

File: src/Message.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ export default class Message extends React.Component {
       <View>
         {this.renderDay()}
         <View style={[styles[this.props.position].container, {
-          marginBottom: this.isSameUser(this.props.currentMessage, this.props.nextMessage) ? 2 : 10,
+          marginBottom: this.isSameUser(this.props.currentMessage, this.props.nextMessage) ? this.props.marginSameUser : this.props.marginDifferentUser,
```
Comment (dgdavid) [inline src/Message.js]: Hi @jayshah123 

I think that this is not related to the original PR. In my opinion, is better to split this changes in two different PR.

File: src/Message.js [hunk 2]
```diff
@@ -119,6 +119,8 @@ Message.defaultProps = {
   renderBubble: null,
   renderDay: null,
   position: 'left',
+  marginSameUser: 6,
```
Comment (dgdavid) [inline src/Message.js]: The original value was `2`. Just for curiosity, Why do you want to change it?

File: src/Message.js (hunk 1 above)
Comment (kfiroo) [inline src/Message.js]: @jayshah123  This is how github works, you stated that the master branch of your fork is the source of the PR so every change in it will be included, you could use a feature branch to achieve what you want

Comment (jayshah123) [inline src/Message.js]: Ok I think I will revert the second commit and that should remove it?

Comment (kfiroo) [inline src/Message.js]: Not sure :)
Give it a try

Comment (jayshah123) [inline src/Message.js]: Rewriting history is Ok? I will close PR, rewrite history and raise a fresh PR(with single commit).

Comment (dgdavid) [inline src/Message.js]: @jayshah123 Don't worry.

You only need to create a new branch in your fork, make the commit on it and create a new PR in the same way you created the previous one.

My recommendation is that this new branch will be created from `master` (in this case).

The Github documentation is better explained that my comment. Here you have some links. I hope it'll be useful for you.

https://guides.github.com/introduction/flow/
https://help.github.com/articles/creating-a-pull-request/

**Comment by jayshah123** [comment]
I'm a novice github user,
messed up this PR by adding a unnecessary commit on master branch.
Will raise a proper PR.

File: src/Message.js (hunk 1 above)
Comment (jayshah123) [inline src/Message.js]: Aha Got it thanks!
